modules/ort/src/main/java/org/eclipse/sw360/antenna/ort/resolver/OrtAnalyzerResultResolver.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Bosch Software Innovations GmbH 2019.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.sw360.antenna.ort.resolver;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.eclipse.sw360.antenna.model.artifact.Artifact;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactFilename;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactHomepage;
+import org.eclipse.sw360.antenna.model.artifact.facts.ArtifactSourceUrl;
+import org.eclipse.sw360.antenna.model.artifact.facts.DeclaredLicenseInformation;
+import org.eclipse.sw360.antenna.util.LicenseSupport;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Optional;
+
+public class OrtAnalyzerResultResolver extends AbstractOrtResultResolver {
+    public OrtAnalyzerResultResolver(JsonNode result) {
+        super(result);
+    }
+
+    @Override
+    public Artifact mapArtifact() {
+        Artifact a  = super.mapArtifact();
+        mapDeclaredLicense().ifPresent(a::addFact);
+        mapFilename().ifPresent(a::addFact);
+        mapHomepage().ifPresent(a::addFact);
+        return a;
+    }
+
+    private Optional<DeclaredLicenseInformation> mapDeclaredLicense() {
+        if (result.get("declared_licenses").isArray()) {
+            Iterator<JsonNode> licenseIterator = result.get("declared_licenses").elements();
+            Collection<String> licenses = new ArrayList<>();
+            while (licenseIterator.hasNext()) {
+                licenses.add(licenseIterator.next().textValue());
+            }
+            return Optional.of(new DeclaredLicenseInformation(LicenseSupport.mapLicenses(licenses)));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<ArtifactSourceUrl> mapSourceUrl() {
+        if (result.has("source_artifact")) {
+            return Optional.of(new ArtifactSourceUrl(result.get("source_artifact").get("url").textValue()));
```

Comment:
this function (and all others with the `if (result.has(...)) { ... } return Optional.empty()` pattern) can be simplified to:
```java
Optioal.ofNullable(result.get("source_artifact"))
     .map(v -> v.get("url").textValue())
     .map(ArtifactSourceUrl::new);
```
see: https://fasterxml.github.io/jackson-databind/javadoc/2.8/com/fasterxml/jackson/databind/JsonNode.html#has(java.lang.String)